### PR TITLE
Update sysctl template remediation to comment parameters from /etc/sysctl.d/*.conf

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Update sysctl template remediation to comment parameters from /etc/sysctl.d/*.conf
  - Comment out any parameters found in /etc/sysctl.d/*.conf files before
remediating to /etc/sysctl.conf.

### CODE REVIEW

**Review by Mab879 — DISMISSED** [review]

**Comment by Mab879** [comment]
Tests pass in a VM. Waiting on OpenShift tests.

**Comment by Mab879** [comment]
/retest

One last time. The logs makes it seems like an Infra issue.

**Comment by ggbecker** [comment]
ocp4 problem reported here as it's unrelated to the PR: https://github.com/ComplianceAsCode/content/issues/8012

**Comment by ggbecker** [comment]
@marcusburghardt if you can please take a look at this a merge if you have no objections.

File: shared/templates/sysctl/bash.template [hunk 1]
```diff
@@ -4,6 +4,20 @@
 # complexity = low
 # disruption = medium
 
+# Comment out any occurrences of {{{ SYSCTLVAR }}} from /etc/sysctl.d/*.conf files
+for f in /etc/sysctl.d/*.conf ; do
+  if [ ! -e "$f" ] ; then
+    continue
+  fi
```
Comment (marcusburghardt) [inline shared/templates/sysctl/bash.template]: @ggbecker this conditional should never be true. We can remove it.

**Review by marcusburghardt — CHANGES_REQUESTED** [review]

File: linux_os/guide/system/permissions/restrictions/poisoning/grub2_page_poison_argument/tests/disabled_d_directory.fail.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+setting_name="kernel.page_poison"
+setting_value="0"
+# sysctl -w "$setting_name=$setting_value"
+if grep -q "^$setting_name" /etc/sysctl.d/99-sysctl.conf; then
+    sed -i "s/^$setting_name.*/$setting_name = $setting_value/" /etc/sysctl.d/99-sysctl.conf
+else
+    echo "$setting_name = $setting_value" >> /etc/sysctl.d/99-sysctl.conf
+fi
+
+sysctl --system
```
Comment (marcusburghardt) [inline linux_os/guide/system/permissions/restrictions/poisoning/grub2_page_poison_argument/tests/disabled_d_directory.fail.sh:23]: This test is failing for me. It changes sysctl parameters but the rule is related to grub configuration. Maybe it is in the wrong folder.

**Review by marcusburghardt — CHANGES_REQUESTED** [review]

**Comment by ggbecker** [comment]
I have relocated the test scenario to another rule. Please check it again

File: linux_os/guide/system/permissions/restrictions/poisoning/grub2_page_poison_argument/tests/disabled_d_directory.fail.sh (hunk 2 above)
Comment (ggbecker) [reply linux_os/guide/system/permissions/restrictions/poisoning/grub2_page_poison_argument/tests/disabled_d_directory.fail.sh]: moved to `sysctl_kernel_randomize_va_space/tests/wrong_value_d_directory.fail.sh`

**Comment by marcusburghardt** [comment]
Now all tests are working as expected in VMs.
The test `SSGTS / Run Tests (pull_request)` is failing due to the container based test environment. Consequently, the sysctl `kernel.randomize_va_space`  cannot be set in runtime. So, we are safe to waive it here.

**Review by marcusburghardt — APPROVED** [review]
Thanks @ggbecker

**Pull request merged**